### PR TITLE
New helper methods for vulnerability detector to query both hotfixes and packages from agents

### PR DESCRIPTION
**Pull request by pereyra-m**
|Related issue|
|---|
|#8555 #8570|

## Description

This PR creates two helpers for vulnerability detector, to retrieve the agents' hotfixes and packages.
They receive all the data sent by Wazuh-DB, and also check for the synchronization message.

## Tests

- Compilation without warnings in every supported platform
  - [x] Linux
- [x] Source installation

- Memory tests for Linux
  - [x] Scan-build report

### CODE REVIEW

**Comment by pereyra-m** [comment]
### Tests

The manual tests confirm the commands are working as expected.
Also, `scan-build` didn't report any bug.

scan_build

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 1]
```diff
@@ -8027,4 +8029,97 @@ void wm_vuldel_truncate_revision(char * revision) {
...
+    return result;
+}
+
+int wm_vuldet_get_software(int agent_id, bool not_triaged, cJSON** requested_items) {
+    int result = OS_INVALID;
+    char query[OS_SIZE_128];
+
+    *requested_items = NULL;
+
+    snprintf(query, OS_SIZE_128, vu_queries[VU_SOFTWARE_GET], agent_id, not_triaged ? "not_triaged" : "triaged");
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8114]: For the cases where the client wants to get all the packages, is a bit confusing the usage of the word `triaged`. And in fact, the Wazuh DB parser does not take into account that keyword.

My suggestion is to only make use of the `not_triaged` keyword and if the client wants to get all the packages, send the command without any parameter.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 2]
```diff
@@ -8027,4 +8029,97 @@ void wm_vuldel_truncate_revision(char * revision) {
...
+    } while (size != OS_INVALID && parse_result == WDBC_DUE);
+
+    if (OS_INVALID == result || FALSE == sync_status) {
+        cJSON_Delete(*j_array);
+    }
+    os_free(response);
+    return result;
+}
+
+int wm_vuldet_get_hotfixes(int agent_id, cJSON** requested_items) {
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8091]: You should add the definition of this function at the beginning of the file to allow usage.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 3]
```diff
@@ -8027,4 +8029,97 @@ void wm_vuldel_truncate_revision(char * revision) {
...
+    if (wm_vuldet_send_wdb(query) != OS_SUCCESS) {
+        mterror(WM_VULNDETECTOR_LOGTAG, VU_SOFTWARE_REQUEST_ERROR, agent_id);
+    } else {
+        result = wm_vuldet_recv_wdb_by_row(requested_items);
+    }
+
+    return result;
+}
+
+int wm_vuldet_get_software(int agent_id, bool not_triaged, cJSON** requested_items) {
```
Comment (DProvinciani) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8108]: DITTO

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 1 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c]: You are right!
Fixed

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 2 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8097]: Definition and doxygen added !

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 3 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8114]: Definition and doxygen added !

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector_db.h [hunk 4]
```diff
@@ -74,6 +74,8 @@ typedef enum vu_query {
     VU_REMOVE_UNUSED_STATES,
     // WAZUH DB REQUESTS
     VU_HOTFIXES_REQUEST,
+    VU_HOTFIXES_GET,
+    VU_SOFTWARE_GET,
```
Comment (palaciosjeremias) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector_db.h:78]: I think this query could now be named VU_PACKAGES_GET to be aligned with the commands names.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 5]
```diff
@@ -8027,4 +8029,103 @@ void wm_vuldel_truncate_revision(char * revision) {
     return;
 }
 
+int wm_vuldet_recv_wdb_by_row(cJSON** j_array) {
+    char* response = NULL;
+    char* payload = NULL;
+    const char* error = NULL;
+    int size;
```
Comment (palaciosjeremias) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8036]: We should initialize size and parse_result.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 6]
```diff
@@ -8027,4 +8029,103 @@ void wm_vuldel_truncate_revision(char * revision) {
...
+        j_object = cJSON_ParseWithOpts(payload, &error, TRUE);
+        if (!j_object) {
+            mdebug1("Invalid JSON syntax in vulnerability detector database request.");
+            mdebug2("JSON error near: %s", error);
+            result = OS_INVALID;
+            break;
+        }
+
+        switch (parse_result) {
+            case WDBC_OK:
```
Comment (palaciosjeremias) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8063]: We can add two comments. One here explaining that WDB_OK represents the end of the stream of items.

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c [hunk 7]
```diff
@@ -8027,4 +8029,103 @@ void wm_vuldel_truncate_revision(char * revision) {
...
+                    } else {
+                        sync_status = FALSE;
+                        result = OS_INVALID;
+                    }
+                } else {
+                    sync_status = FALSE;
+                    result = OS_INVALID;
+                }
+                break;
+            case WDBC_DUE:
```
Comment (palaciosjeremias) [inline src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8081]: And one here explaining that a new itemwas received and there are pending ones.

**Review by palaciosjeremias — APPROVED** [review]
LGTM

**Review by DProvinciani — APPROVED** [review]
Looks good!

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 5 above)
Comment (pereyra-m) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8036]: Done!

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector_db.h (hunk 4 above)
Comment (DProvinciani) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector_db.h:78]: Fixed

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 5 above)
Comment (DProvinciani) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8036]: Fixed

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 6 above)
Comment (DProvinciani) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8063]: Fixed

File: src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c (hunk 7 above)
Comment (DProvinciani) [reply src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c:8081]: Fixed

**Review by palaciosjeremias — APPROVED** [review]

**Pull request merged**